docs/proposals/deployment-insanity.md

```diff
@@ -0,0 +1,29 @@
+Dropping into a doc so we have a place to argue.
+
+We want to enabling deployments(Ds) and someday we want feature parity with DCs and then we'll likely retire DCs.
+
+This means that we need to make sure that no D or DC have conflicting names.  The simplest way to 
+do this is to have the two resources cohabitate (live in the same directory in etcd).  However,
+they are different than our other migrating/cohabitating resources like replicationcontrollers and 
+replicasets.  RC and RS controllers are close enough to parity that we're going to be able to run
+a single controller to manage both.  D and DC controllers are not close enough to do that.
+
+Since we'll be running two controllers, we'll need to have the controllers know which resources under
+the D or DC endpoint they should be managing.  We don't yet have controllerRefs and we're unlikely to 
+pick them in the next couple days.  The fastest way that I see to do this is:
+ 1. Extend `StorageFactory` to allow the storage encoding version for cohabitating resources vary based
+ on which resource is being requested.
+ 2. Update conversions from `DC.v1->D.__internal.extensions` and `D.v1beta1.extensions->DC.__internal` to
+ add an annotation: `k8s.io/original-kind=GroupKind`.
```

Comment:
> And actually store it? Seems like that would make a migration more difficult since we'd have to fix up the serialized bits. Could be done, but it doesn't seem like it makes things easier.

How do we guarantee that the annotation won't be stored? Clients will need to see it in order to identify the object, won't the annotation persist when they send an update back to the storage?